.github/workflows/release.yml

```diff
@@ -340,6 +341,40 @@ jobs:
                   cd test
                   pnpm moonwall test dev_tanssi
 
+    typescript-tests-flashbox:
+        runs-on: ubuntu-latest
+        needs: ["set-tags", "build"]
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v3
+              with:
+                  ref: ${{ needs.set-tags.outputs.git_ref }}
+            - name: Pnpm
+              uses: pnpm/action-setup@v2
+              with:
+                  version: 8
+            - name: "Download binaries"
+              uses: actions/download-artifact@v3.0.2
+              with:
+                  name: binaries
+                  path: target/release
+            - name: "Setup Node"
+              uses: actions/setup-node@v3
+              with:
+                  node-version: 20.x
+                  cache: "pnpm"
+            - name: "Generate typescript api"
+              run: |
+                  chmod uog+x target/release/tanssi-node
+                  cd typescript-api
+                  pnpm install
+                  pnpm create-local-interfaces
```

Comment:
How does typescript-api support multiple runtimes? As I understand it, it spawns a node and gets the list of pallets and extrinsics. So I guess it will use the dancebox typescript interface for flashbox?

Reply:
Yeah, as it is now it's using dancebox's TS interface. We could [modify the generate script](https://github.com/moondance-labs/tanssi/blob/master/typescript-api/scripts/runtime-upgrade.ts#L35) to accept an arg so it know what interface to generate, but I think that using the dancebox one should work for what we need. I might be wrong though

Reply:
So moonbeam has a separate typescript-api for each runtime, but I don't know how they import that in tests: https://github.com/moonbeam-foundation/moonbeam/tree/master/typescript-api/src

Adding flashbox to the typescript-api is a one line change here:
https://github.com/moondance-labs/tanssi/blob/c43bce59e2258fe244ccb55002156f62e047f2f6/typescript-api/scripts/runtime-upgrade.ts#L9

But I don't know how to ensure that it is imported properly in tests...

Reply:
yes I will make sure I generate a separate one for flashbox, will do in a separate PR

Reply:
I was also looking into it and arrived at the same question. I think there's no easy way of choosing what interface to import on tests.
I guess it depends on what the future of flashbox looks like. Will it at some point differ enough from dancebox that it require its own interface? Or is the plan to keep it as close as posible to dancebox (including the same pallet indices)?
(cc @girazoki)

Reply:
I agree with @girazoki that we can handle this in a separate pull request. The way we're doing things now will be fine until we decide to add a pallet to flashbox but not to dancebox, and that might not even happen.

Reply:
I've gone and added the flashbox interface. Still not sure how to use it on the test environment

Reply:
since the tests are common, let's use the dancebox one